### PR TITLE
Added Error Classes for RoutingApi

### PR DESCRIPTION
**Pull request by eifinger**
For downstream usage of this lib it would be helpful to get more detailed Exception to display to the user what went wrong.

This is a proposed way of handling that problem, it could be implemented in the same way for the other APIs.

What do you think?

## Example for publicTransport RouteNotFound:

In the current implementation I have to parse the output of HEREError.message() and look for ``NGEO_ERROR_GRAPH_DISCONNECTED``

Then I can write a better error method for the users telling them that the actual error was that HERE couldn't find a route using public Transport.
It would be easier to just check if the returned Error is of type NoRouteFoundError.

### CODE REVIEW

**Comment by eifinger** [comment]
Seems to work now.
I will also add some additional tests tomorrow.

**Comment by abdullahselek** [comment]
Awesome, adding tests would be great 👍

**Comment by eifinger** [comment]
Added the additional tests.
I could not produce any errors for ``WaypointNotFoundError``, ``LinkIdNotFoundError`` and ``RouteNotReconstructedError`` when using the api manually.

File: herepy/routing_api.py [hunk 1]
```diff
@@ -178,3 +178,89 @@ def truck_route(self,
...
+
+
+
+class InvalidCredentialsError(HEREError):
+
+    """Invalid Credentials Error Type.
+
+    This error is returned if the specified token was invalid or no contract
+    could be found for this token.
+
```
Comment (abdullahselek) [inline herepy/routing_api.py]: It's minor thing but can you delete these empty lines before closing comment?

Comment (abdullahselek) [reply herepy/routing_api.py]: Same each for new error class.

File: herepy/routing_api.py [hunk 2]
```diff
@@ -178,3 +178,89 @@ def truck_route(self,
...
+    """
+
+
+class InvalidInputDataError(HEREError):
```
Comment (abdullahselek) [inline herepy/routing_api.py]: If we can use new error classes in general I think it's better to locate them in error.py.

File: herepy/routing_api.py (hunk 1 above)
Comment (eifinger) [reply herepy/routing_api.py]: Done

File: herepy/routing_api.py (hunk 2 above)
Comment (eifinger) [reply herepy/routing_api.py]: I think the APIs are developed by completely separate teams and they are behaving extremely different in terms of communication patterns, documentation and so on.

The Routing API is the only API which offers Error Types as far as I could find in the documentation: https://developer.here.com/documentation/routing/topics/resource-types-error.html

The Places API for example does not use schemas etc but returns
```json
{
    "status": 401,
    "message": "Invalid app_id app_code combination"
}
````

As the APIs are clearly disconnected I would also separate the Error Classes.

Comment (abdullahselek) [reply herepy/routing_api.py]: Ok, no worries.

**Comment by abdullahselek** [comment]
All looks good, thanks 👍

**Pull request merged**